### DISCUSSION TITLE
How to get a document using useDocument

### DISCUSSION THREAD

**Discussion by joseluisgs** [vuejs/vuefire#1268] [Questions and Help]
Hi @posva and other Vuefire Team

I have the following code
```js
const myBoardRef = doc(boardsCollection, uid)
const myBoard = await getDoc(myBoardRef)
console.log(myBoard.data())
```

I would like to use usedocument, to obtain a real Time ref but I have problemst to get the Document

```js
const myBoard = useDocument(doc(boardsCollection, uid))
```

Bu I don't know how I can get the value, I use .value but always is undefined. I follow: https://vuefire.vuejs.org/guide/realtime-data.html

Could you help me to solve it?

Thank you

**Comment by posva** [comment]
You can await the promise from https://vuefire.vuejs.org/guide/realtime-data.html#subscription-state

note that useDocument is a composable and meant to be used within setup of components

**Comment by joseluisgs** [comment]
Hi @posva thank you again.

I have a pinia store, and I have a computed property I would like to obtain the same result, I mean, get a subscription in this document (board), but I can not I don't know why.
I have the raw value with the data of the board but always is undefined when Iill try to get the value

Please could you help me with a example that simulate my use case

Thank you

PS: Eduardo, feliz navidad. Y perdona que te haga trabajar, contesta cuando puedas que bastante ya haces :)

**Reply by posva** [reply]
Hello! Sorry for the late reply, I was going through discussions again. I think you were facing a bug in the past and that this should be fixed now. Note the data starts as empty until Firebase populates it. If you still have a bug, please open an issue with a boiled down reproduction, I added starters for it.

PS: Gracias!
